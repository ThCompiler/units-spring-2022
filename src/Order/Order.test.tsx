jest.mock('../utils/getDate');
import React from 'react';
import {configure, mount, render, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {OrderComponent} from './Order';
import {getDate} from '../utils/getDate';

configure({ adapter: new Adapter() });

beforeEach(() => {
	(getDate as jest.Mock).mockReturnValue('2 апреля 2022 год');
});

afterEach(() => {
	jest.clearAllMocks();
});

describe.each([ [{
	date: 2,
	id: 3,
	items: ['d', 'k'],
	shop: 'dore',
}, 'with items'],
[{
	date: 2,
	id: 3,
	items: [],
	shop: 'dore',
}, 'without items'],
[{
	date: 2,
	items: ['d', 'k'],
	shop: 'dore',
}, 'no id'],
]) ( 'with call getDate', (a, type: string)=> {

	it(type + ' : shallow', () => {
		const wrapper = shallow(<OrderComponent order={a}/>);

		expect(getDate).toBeCalledTimes(1);
		expect(getDate).toBeCalledWith(a.date);
		expect(wrapper).toMatchSnapshot();
	});

	it(type + ' : mount', () => {
		const wrapper = mount(<OrderComponent order={a}/>);

		expect(getDate).toBeCalledTimes(1);
		expect(getDate).toBeCalledWith(a.date);
		expect(wrapper).toMatchSnapshot();
	});

	it(type + ' : render', () => {
		const wrapper = render(<OrderComponent order={a}/>);

		expect(getDate).toBeCalledTimes(1);
		expect(getDate).toBeCalledWith(a.date);
		expect(wrapper).toMatchSnapshot();
	});
});

describe.each([
	[{
		id: 3,
		items: ['d', 'k'],
		shop: 'dore',
	}, 'no date'],
	[{
		id: 3,
		date: 2,
		items: ['d', 'k'],
	}, 'no shop'],
]) ( 'without call getDate', (a, type: string)=> {

	it(type + ' : shallow', () => {
		const wrapper = shallow(<OrderComponent order={a}/>);

		expect(getDate).toBeCalledTimes(0);
		expect(wrapper).toMatchSnapshot();
	});

	it(type + ' : mount', () => {
		const wrapper = mount(<OrderComponent order={a}/>);

		expect(wrapper).toMatchSnapshot();
	});

	it(type + ' : render', () => {
		const wrapper = render(<OrderComponent order={a}/>);

		expect(getDate).toBeCalledTimes(0);
		expect(wrapper).toMatchSnapshot();
	});
});
