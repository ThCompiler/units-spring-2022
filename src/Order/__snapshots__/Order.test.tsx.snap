// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with call getDate no id : mount 1`] = `
<OrderComponent
  order={
    Object {
      "date": 2,
      "items": Array [
        "d",
        "k",
      ],
      "shop": "dore",
    }
  }
>
  <div
    className="Order"
  >
    <div
      className="Order-header"
    >
      <span
        className="Order-shop"
      >
        dore
      </span>
      <span
        id="date"
      >
        2 апреля 2022 год
      </span>
    </div>
    <div
      className="Order-items"
    >
      <div
        className="Order-item"
        key="0"
      >
        d
      </div>
      <div
        className="Order-item"
        key="1"
      >
        k
      </div>
    </div>
  </div>
</OrderComponent>
`;

exports[`with call getDate no id : render 1`] = `
<div
  class="Order"
>
  <div
    class="Order-header"
  >
    <span
      class="Order-shop"
    >
      dore
    </span>
    <span
      id="date"
    >
      2 апреля 2022 год
    </span>
  </div>
  <div
    class="Order-items"
  >
    <div
      class="Order-item"
    >
      d
    </div>
    <div
      class="Order-item"
    >
      k
    </div>
  </div>
</div>
`;

exports[`with call getDate no id : shallow 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      dore
    </span>
    <span
      id="date"
    >
      2 апреля 2022 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      d
    </div>
    <div
      className="Order-item"
      key="1"
    >
      k
    </div>
  </div>
</div>
`;

exports[`with call getDate with items : mount 1`] = `
<OrderComponent
  order={
    Object {
      "date": 2,
      "id": 3,
      "items": Array [
        "d",
        "k",
      ],
      "shop": "dore",
    }
  }
>
  <div
    className="Order"
  >
    <div
      className="Order-header"
    >
      <span
        className="Order-shop"
      >
        dore
      </span>
      <span
        id="date"
      >
        2 апреля 2022 год
      </span>
    </div>
    <div
      className="Order-items"
    >
      <div
        className="Order-item"
        key="0"
      >
        d
      </div>
      <div
        className="Order-item"
        key="1"
      >
        k
      </div>
    </div>
  </div>
</OrderComponent>
`;

exports[`with call getDate with items : render 1`] = `
<div
  class="Order"
>
  <div
    class="Order-header"
  >
    <span
      class="Order-shop"
    >
      dore
    </span>
    <span
      id="date"
    >
      2 апреля 2022 год
    </span>
  </div>
  <div
    class="Order-items"
  >
    <div
      class="Order-item"
    >
      d
    </div>
    <div
      class="Order-item"
    >
      k
    </div>
  </div>
</div>
`;

exports[`with call getDate with items : shallow 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      dore
    </span>
    <span
      id="date"
    >
      2 апреля 2022 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      d
    </div>
    <div
      className="Order-item"
      key="1"
    >
      k
    </div>
  </div>
</div>
`;

exports[`with call getDate without items : mount 1`] = `
<OrderComponent
  order={
    Object {
      "date": 2,
      "id": 3,
      "items": Array [],
      "shop": "dore",
    }
  }
>
  <div
    className="Order"
  >
    <div
      className="Order-header"
    >
      <span
        className="Order-shop"
      >
        dore
      </span>
      <span
        id="date"
      >
        2 апреля 2022 год
      </span>
    </div>
    <div
      className="Order-items"
    >
      <div>
        Заказ пуст...
      </div>
    </div>
  </div>
</OrderComponent>
`;

exports[`with call getDate without items : render 1`] = `
<div
  class="Order"
>
  <div
    class="Order-header"
  >
    <span
      class="Order-shop"
    >
      dore
    </span>
    <span
      id="date"
    >
      2 апреля 2022 год
    </span>
  </div>
  <div
    class="Order-items"
  >
    <div>
      Заказ пуст...
    </div>
  </div>
</div>
`;

exports[`with call getDate without items : shallow 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      dore
    </span>
    <span
      id="date"
    >
      2 апреля 2022 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div>
      Заказ пуст...
    </div>
  </div>
</div>
`;

exports[`without call getDate no date : mount 1`] = `
<OrderComponent
  order={
    Object {
      "id": 3,
      "items": Array [
        "d",
        "k",
      ],
      "shop": "dore",
    }
  }
/>
`;

exports[`without call getDate no date : render 1`] = `null`;

exports[`without call getDate no date : shallow 1`] = `""`;

exports[`without call getDate no shop : mount 1`] = `
<OrderComponent
  order={
    Object {
      "date": 2,
      "id": 3,
      "items": Array [
        "d",
        "k",
      ],
    }
  }
/>
`;

exports[`without call getDate no shop : render 1`] = `null`;

exports[`without call getDate no shop : shallow 1`] = `""`;
